src/pubsub/shared.rs

```diff
@@ -107,19 +119,18 @@ impl ConnectionManager {
     ) -> (RouteTask<T>, WriteTask<T>) {
         let (tx, rx) = mpsc::channel(self.notification_buffer_per_task);
 
-        let (gone_tx, gone_rx) = oneshot::channel();
+        let tasks = self.root_tasks.child();
```

Comment:
connections each get their own task set, which work is spawned onto. this work is automatically cancelled when the connection goes away 😎